packages/epo-widget-lib/src/widgets/SourceSelector/SourceSelector.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  FunctionComponent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useState,
+} from "react";
+import { Alert, Source } from "@/types/astro";
+import { Image } from "@rubin-epo/epo-react-lib/dist/types/image";
+import { useTranslation } from "react-i18next";
+import { IconComposer } from "@rubin-epo/epo-react-lib";
+import * as Styled from "./styles";
+import Points from "./Points";
+import Message from "./Message";
+import { getLinearScale } from "./utilities";
+
+interface BlinkConfig {
+  autoplay?: boolean;
+  loop?: boolean;
+  duration?: number;
+}
+
+interface Elapsed {
+  day: number;
+  hour: number;
+}
+
+interface SourceSelectorProps {
+  width?: number;
+  height?: number;
+  sources: Source[];
+  alerts?: Alert[];
+  selectedSource?: Source[];
+  images: Image[];
+  selectionCallback?: (data: Source[]) => void;
+  blinkConfig?: BlinkConfig;
+  color?: string;
+}
+
+const SourceSelector: FunctionComponent<SourceSelectorProps> = ({
+  width = 600,
+  height = 600,
+  selectedSource: preSelectedSource = [],
+  sources,
+  alerts,
+  images,
+  selectionCallback,
+  blinkConfig,
+  color,
+}) => {
+  const [selectedSource, setSelectedSource] =
+    useState<Source[]>(preSelectedSource);
+  const [isLoaded, setLoaded] = useState(false);
+  const [message, setMessage] = useState<ReactNode>();
+  const [isMessageVisible, setMessageVisible] = useState(false);
+  const [elapsed, setElapsed] = useState<Elapsed | null>(null);
+  const { t } = useTranslation();
+  const svgId = "sourceSelectorWidget";
+
+  useEffect(() => {
+    selectionCallback && selectionCallback(selectedSource);
+  }, [selectedSource]);
+
+  const findData = (data: Source[], id: string, type: string) =>
+    data.filter((d) => d.id === id && d.type === type);
+
+  const handleClick = (event: MouseEvent<SVGElement>) => {
```

Comment:
to replace the D3 click functionality, the entire `svg` has an onClick handler attached which then checks for the `id` and `type` data attributes and matches them to sources and figures out if a source was clicked and if that source has already been clicked.